sky/packages/sky/lib/src/rendering/box.dart

```diff
@@ -100,80 +117,65 @@ class BoxConstraints extends Constraints {
-                              minHeight: minHeight,
-                              maxHeight: maxHeight);
-  }
-
-  BoxConstraints applyHeight(double height) {
+  /// Returns a new box constraints with a tight height as close to the given height as possible while still respecting the original box constraints
+  BoxConstraints tightenHeight(double height) {
     return new BoxConstraints(minWidth: minWidth,
                               maxWidth: maxWidth,
                               minHeight: math.max(math.min(maxHeight, height), minHeight),
                               maxHeight: math.max(math.min(maxHeight, height), minHeight));
   }
 
-  BoxConstraints applyMinHeight(double newMinHeight) {
-    return new BoxConstraints(minWidth: minWidth,
-                              maxWidth: maxWidth,
-                              minHeight: math.max(minHeight, newMinHeight),
-                              maxHeight: math.max(maxHeight, newMinHeight));
-  }
-
-  BoxConstraints applyMaxHeight(double newMaxHeight) {
-    return new BoxConstraints(minWidth: minWidth,
-                              maxWidth: maxWidth,
-                              minHeight: minHeight,
-                              maxHeight: math.min(maxHeight, newMaxHeight));
-  }
-
+  /// Returns a box constraints with the same width constraints but with unconstrainted height
   BoxConstraints widthConstraints() => new BoxConstraints(minWidth: minWidth, maxWidth: maxWidth);
 
+  /// Returns a box constraints with the same height constraints but with unconstrainted width
   BoxConstraints heightConstraints() => new BoxConstraints(minHeight: minHeight, maxHeight: maxHeight);
 
-  final double minWidth;
-  final double maxWidth;
-  final double minHeight;
-  final double maxHeight;
-
+  /// Returns the width that both satisfies the constraints and is as close as possible to the given width
   double constrainWidth([double width = double.INFINITY]) {
     return clamp(min: minWidth, max: maxWidth, value: width);
   }
 
+  /// Returns the height that both satisfies the constraints and is as close as possible to the given height
   double constrainHeight([double height = double.INFINITY]) {
     return clamp(min: minHeight, max: maxHeight, value: height);
   }
 
+  /// Returns the size that both satisfies the constraints and is as close as possible to the given size
   Size constrain(Size size) {
     Size result = new Size(constrainWidth(size.width), constrainHeight(size.height));
     if (size is _DebugSize)
       result = new _DebugSize(result, size._owner, size._canBeUsedByParent);
     return result;
   }
+
+  /// The biggest size that satisifes the constraints
   Size get biggest => new Size(constrainWidth(), constrainHeight());
+
+  /// The smallest size that satisfies the constraints
   Size get smallest => new Size(constrainWidth(0.0), constrainHeight(0.0));
 
+  /// Whether either the width or the height is unbounded
```

Comment:
actually isInfinite only returns true if they're both infinite. And as far as I can tell it's only used in two asserts? Maybe we can drop this. It's confusing that it doesn't work quite like size.isInfinite.